web-app/src/app/screens/CompleteRegistration.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Container from '@mui/material/Container';
+import Typography from '@mui/material/Typography';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import { getAuth, updateProfile } from 'firebase/auth';
+import 'firebase/firestore';
+import {
+  Alert,
+  Box,
+  Checkbox,
+  CssBaseline,
+  FormControlLabel,
+} from '@mui/material';
+import { useAppDispatch } from '../hooks';
+import { refreshUserInformation } from '../store/profile-reducer';
+import { useNavigate } from 'react-router-dom';
+import { ACCOUNT_TARGET } from '../constants/Navigation';
+
+export default function CompleteRegistration(): React.ReactElement {
+  const auth = getAuth();
+  const user = auth.currentUser;
+  const dispatch = useAppDispatch();
+  const navigateTo = useNavigate();
+
+  const CompleteRegistrationSchema = Yup.object().shape({
+    fullname: Yup.string().required('Your full name is required.'),
+    requiredCheck: Yup.boolean().oneOf([true], 'This field must be checked'),
+    agreeToTerms: Yup.boolean()
+      .required('You must accept the terms and conditions.')
+      .isTrue('You must accept the terms and conditions.'),
+    agreeToPrivacyPolicy: Yup.boolean()
+      .required('You must agree to the privacy policy.')
+      .isTrue('You must agree to the privacy policy.'),
+  });
+
+  const formik = useFormik({
+    initialValues: {
+      fullname: '',
+      organizationName: '',
+      receiveAPIannouncements: false,
+      agreeToTerms: false,
+      agreeToPrivacyPolicy: false,
+    },
+    validationSchema: CompleteRegistrationSchema,
+    onSubmit: async (values) => {
+      if (user != null) {
+        await updateProfile(user, {
+          displayName: values?.fullname,
+        });
+        dispatch(refreshUserInformation(user));
```

Comment:
This operation should be abstracted from the component. The update of the profile should be in the saga layer not the component.